dali/base/dasds.cpp

```diff
@@ -5753,52 +5767,52 @@ class CStoreHelper : implements IStoreHelper, public CInterface
         unsigned edition = storeInfo.edition;
         unsigned newEdition = nextEditionN(edition);
         bool savesCompleted = false;
+
+        OwnedIFile iXmlFileTmpStore;
         try
         {
-            // Save store as XML
-            StringBuffer tmpStoreName;
-            constexpr const char *temporaryXmlFileSaveName = "dali_store_tmp.xml";
-            OwnedIFileIO iXmlFileIOTmpStore = createUniqueFile(location, temporaryXmlFileSaveName, NULL, tmpStoreName);
-            OwnedIFile iXmlFileTmpStore = createIFile(tmpStoreName);
-            unsigned xmlCrc{0};
-            try
-            {
-                xmlCrc = saveStoreToFile(root, iXmlFileIOTmpStore, StoreFormat::XML);
-            }
-            catch (IException *e)
-            {
-                StringBuffer errMsg;
-                EXCLOG(e, errMsg.append("Failed to save XML temporary store to : ").append(temporaryXmlFileSaveName).str());
-                e->Release();
-                iXmlFileTmpStore->remove();
-                return;
-            }
-
-            // Save store as binary
             OwnedIFile iBinaryFileTmpStore;
+            unsigned xmlCrc{0};
             unsigned binaryCrc{0};
-            unsigned *binaryCrcPtr{nullptr};
+
+            // Execute save operations
             if (saveBinary)
             {
-                StringBuffer tmpStoreName;
-                constexpr const char *temporaryBinaryFileSaveName = "dali_store_tmp.bin";
-                OwnedIFileIO iBinaryFileIOTmpStore = createUniqueFile(location, temporaryBinaryFileSaveName, NULL, tmpStoreName);
-                iBinaryFileTmpStore.setown(createIFile(tmpStoreName));
+                std::future<IFile *> xmlStoreSavedFuture;
+                if (saveAsync)
+                {
+                    // Save the XML in the background
+                    auto asyncXmlSaveFunc = [this, &xmlCrc, root]() -> IFile *
+                    {
+                        LOG(MCdebugProgress, "Saving XML store asynch");
```

Comment:
picky: saveStoreToFile will alread logs "Saving store to XML", and will be obvious from logging that on a different TID, so prob. wouldn't bother with this line.